lib/dependable.rb

```diff
@@ -0,0 +1,19 @@
+module Dependable
+  extend ActiveSupport::Concern
+
+  included do
+    has_many :records_of_dependent_editions, class_name: 'EditionDependency', as: :dependable, dependent: :destroy
+    has_many :dependent_editions, through: :records_of_dependent_editions, source: :edition
+  end
+
+  def republish_dependent_editions
+    dependent_editions.each { |e| Whitehall::PublishingApi.republish(e) }
+  end
+
+  def destroy_records_of_dependent_editions
```

Comment:
in the interest of not adding more lines to `Edition`, and following the pattern of concerns: `Edition::Identifiable`, `Edition::Workflow`, `Edition::Translatable` etc. i thought of keeping this method here.